networkx/classes/graph.py

```diff
@@ -606,67 +606,52 @@ def remove_nodes_from(self, nodes):
             except KeyError:
                 pass
 
-    def nodes_iter(self, data=False):
-        """Return an iterator over the nodes.
+    def nodes(self, data=False):
+        """Returns an iterator over the nodes.
 
         Parameters
         ----------
-        data : boolean, optional (default=False)
-               If False the iterator returns nodes.  If True
-               return a two-tuple of node and node data dictionary
+        data : boolean
+               If ``False``, the iterator returns nodes.  If ``True``,
+               the iterator return a two-tuple of node and node data
+               dictionary.
 
         Returns
         -------
-        niter : iterator
-            An iterator over nodes.  If data=True the iterator gives
-            two-tuples containing (node, node data, dictionary)
+        iterator
+            An iterator over nodes, or if ``data`` is ``True``, an
+            iterator over two-tuples of the form ``(node, node data
+            dictionary)``.
 
         Notes
         -----
-        If the node data is not required it is simpler and equivalent
-        to use the expression 'for n in G'.
-
-        >>> G = nx.Graph()   # or DiGraph, MultiGraph, MultiDiGraph, etc
-        >>> G.add_path([0,1,2])
+        If the node data is not required, it is simpler and equivalent
+        to use the expression ``for n in G``, or ``list(G)``.
 
         Examples
         --------
-        >>> G = nx.Graph()   # or DiGraph, MultiGraph, MultiDiGraph, etc
-        >>> G.add_path([0,1,2])
+        There are two simple ways of getting a list of all nodes in the graph::
 
-        >>> [d for n,d in G.nodes_iter(data=True)]
-        [{}, {}, {}]
-        """
-        if data:
-            return iter(self.node.items())
-        return iter(self.node)
+            >>> import networkx as nx
+            >>> G = nx.Graph()
+            >>> G.add_nodes_from(range(3))
+            >>> list(G.nodes())
+            [0, 1, 2]
+            >>> list(G)
+            [0, 1, 2]
 
-    def nodes(self, data=False):
-        """Return a list of the nodes in the graph.
-
-        Parameters
-        ----------
-        data : boolean, optional (default=False)
-               If False return a list of nodes.  If True return a
-               two-tuple of node and node data dictionary
+        To get the node data along with the nodes::
 
-        Returns
-        -------
-        nlist : list
-            A list of nodes.  If data=True a list of two-tuples containing
-            (node, node data dictionary).
+            >>> G = nx.Graph()
+            >>> G.add_nodes_from(range(3))
+            >>> G.node[0]['foo'] = 'bar'
```

Comment:
Why are we changing the examples here? We don't need to repeat the G=nx.Graph() and I think it woud be good to keep the example of adding attributes as part of the G.add_node() syntax.  Perhaps we could use:

    >>> G.add_node(1, time='5pm')
    >>> G.node[0]['foo'] = 'bar'
    >>> list(G.nodes(data=True)
    [(0, {'foo': 'bar'}, (1, {'time': '5pm'}), (2, {})]